src/Form/PrepaidBalanceReportsDownloadForm.php

```diff
@@ -0,0 +1,230 @@
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_m10n\Form;
+
+use Apigee\Edge\Exception\ClientErrorException;
+use Drupal;
+use Drupal\apigee_m10n\ApigeeSdkControllerFactoryInterface;
+use Drupal\apigee_m10n\MonetizationInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Provides a form to to download prepaid balances.
+ */
+class PrepaidBalanceReportsDownloadForm extends FormBase {
+
+  /**
+   * The Apigee Monetization base service.
+   *
+   * @var \Drupal\apigee_m10n\MonetizationInterface
+   */
+  protected $monetization;
+
+  /**
+   * The Apigee SDK controller factory.
+   *
+   * @var \Drupal\apigee_m10n\ApigeeSdkControllerFactoryInterface
+   */
+  protected $sdkControllerFactory;
+
+  /**
+   * The user from route.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $user;
+
+  /**
+   * An array of supported currencies.
+   *
+   * @var array
+   */
+  protected $supportedCurrencies;
+
+  /**
+   * An array of billing documents.
+   *
+   * @var array
+   */
+  protected $billingDocuments;
+
+  /**
+   * PrepaidBalancesDownloadForm constructor.
+   *
+   * @param \Drupal\apigee_m10n\ApigeeSdkControllerFactoryInterface $sdk_controller_factory
+   *   The SDK controller factory.
+   * @param \Drupal\apigee_m10n\MonetizationInterface $monetization
+   *   The Apigee Monetization base service.
+   */
+  public function __construct(ApigeeSdkControllerFactoryInterface $sdk_controller_factory, MonetizationInterface $monetization) {
+    $this->monetization = $monetization;
+    $this->sdkControllerFactory = $sdk_controller_factory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('apigee_m10n.sdk_controller_factory'),
+      $container->get('apigee_m10n.monetization')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'previous_balances_reports_download_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, UserInterface $user = NULL, array $supported_currencies = [], array $billing_documents = []) {
+    $this->user = $user;
```

Comment:
Should this be stored in `$form_state` instead of a class variable?

Reply:
Fixed.